drivers/dac/dac_ad569x.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2024 Jan Kubiznak <jan.kubiznak@deveritec.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/dac.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(dac_ad569x, CONFIG_DAC_LOG_LEVEL);
+
+#define DAC_AD569X_MAX_CHANNEL 0
+
+#define AD569X_CTRL_GAIN(x)  (FIELD_PREP(BIT(11), x))
+#define AD569X_CTRL_REF(x)   (FIELD_PREP(BIT(12), x))
+#define AD569X_CTRL_PD(x)    (FIELD_PREP(BIT_MASK(2) << 13, x))
+#define AD569X_CTRL_RESET(x) (FIELD_PREP(BIT(15), x))
+
+enum ad569x_command {
+	AD569X_CMD_WRITE = 0x10,
+	AD569X_CMD_UPDATE = 0x20,
+	AD569X_CMD_WRITE_AND_UPDATE = 0x30,
+	AD569X_CMD_CONFIGURE = 0x40,
+};
```

Comment:
I would suggest using defines here and `uint8_t` as type, enum is useful if you need to do a `switch` over it and like the compiler to check that you are hitting all cases but this is not what you are doing here and for register access it just looks off.